homeassistant/components/media_source/local_source.py

```diff
@@ -112,11 +112,15 @@ def _build_item_response(self, source_dir_id: str, path: Path, is_child=False):
         if is_dir:
             title += "/"
 
+        media_class = MEDIA_CLASS_MAP.get(
+            mime_type and mime_type.split("/")[0], MEDIA_CLASS_DIRECTORY
+        )
+
         media = BrowseMediaSource(
             domain=DOMAIN,
             identifier=f"{source_dir_id}/{path.relative_to(self.hass.config.path('media'))}",
-            media_class=MEDIA_CLASS_DIRECTORY,
-            media_content_type="directory",
+            media_class=media_class,
+            media_content_type=mime_type or MEDIA_CLASS_DIRECTORY,
```

Comment:
```suggestion
            media_content_type=mime_type,
```
In that case, what about just keeping it `None`?